### PR TITLE
Feature/preview

### PR DESCRIPTION
**Pull request by Andries-Smit**
* Add web modeler preview
* Move default logic to display component
* Display badge when empty
* Convert CRLF to LF

### CODE REVIEW

File: src/Badge.webmodeler.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        const css = require("./ui/Badge.css");
+        this.addPreviewStyle(css, "widget-badge");
+    }
+
+    render() {
+        return createElement(Overlay, {},
+            createElement(Badge, this.transformProps(this.props))
+        );
+    }
+    private transformProps(props: BadgeContainerProps): BadgeProps {
```
Comment (acellam) [inline src/Badge.webmodeler.ts]: space missing between function declarations. Put space between `render` and `transformProps`

File: src/components/Overlay.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                left: 0,
+                position: "absolute",
+                top: 0,
+                width: "100%",
+                zIndex: 10
+            }
+        })
+    );
+
+const preventEvent = (e: SyntheticEvent<any>) => {
```
Comment (acellam) [inline src/components/Overlay.ts]: Remove `any`

File: src/components/__tests__/Badge.spec.ts [hunk 3]
```diff
@@ -1,86 +1,106 @@
...
+            DOM.span(
+                {
+                    className: ("widget-badge badge label-default"),
+                    onClick: jasmine.any(Function) as any,
+                    style: badgeProps.style
+                }, jasmine.any(String))
+        );
+    });
+
+    it("should show no value when no value nor default value provided", () => {
```
Comment (acellam) [inline src/components/__tests__/Badge.spec.ts]: change `nor` to `or`

File: src/components/__tests__/Badge.spec.ts [hunk 4]
```diff
@@ -1,86 +1,106 @@
...
+    });
+
+    it("should show default value when no value is provided", () => {
+        const defaultValue = "default";
+        const badge = createBadge({ badgeType: "label", value: undefined, defaultValue });
+
+        expect(badge.text()).toBe(defaultValue);
+    });
+
+    it("should show no value when no value nor default value provided", () => {
```
Comment (acellam) [inline src/components/__tests__/Badge.spec.ts]: use `or` and not `nor`

File: src/components/__tests__/Badge.spec.ts (hunk 3 above)
Comment (McDoyen) [reply src/components/__tests__/Badge.spec.ts]: Done

File: src/components/__tests__/Badge.spec.ts (hunk 4 above)
Comment (McDoyen) [reply src/components/__tests__/Badge.spec.ts]: Done

File: src/components/Overlay.ts (hunk 2 above)
Comment (Andries-Smit) [reply src/components/Overlay.ts]: done

**Pull request merged**